src/App.tsx

```diff
@@ -35,47 +36,110 @@ export function getReorderedGoods(
+  }
+
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
 export const App: React.FC = () => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [sortType, setSortType] = useState(SortType.NONE);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [isReversed, setIsReversed] = useState(false);
+
+  const reorderedGoods = getReorderedGoods(goodsFromServer, {
+    sortType,
+    isReversed,
+  });
+
+  //Just playing around with classnames
+  const basicClassesForButton = cn('button');
+
+  const buttonSortAlphabet = cn(basicClassesForButton, 'is-info', {
+    'is-light': sortType !== SortType.ALPHABET,
+  });
+
+  const buttonSortLength = cn(basicClassesForButton, 'is-success', {
+    'is-light': sortType !== SortType.LENGTH,
+  });
+
+  const buttonReverse = cn(basicClassesForButton, 'is-warning', {
+    'is-light': !isReversed,
+  });
+
+  const buttonReset = cn(basicClassesForButton, 'is-danger', {
+    'is-light': sortType === SortType.NONE && !isReversed,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={buttonSortAlphabet}
+          onClick={() => {
+            setSortType(SortType.ALPHABET);
+          }}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={buttonSortLength}
+          onClick={() => {
+            setSortType(SortType.LENGTH);
+          }}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={buttonReverse}
+          onClick={() => {
+            setIsReversed(!isReversed);
+          }}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortType !== 0 || isReversed) && (
+          <button
+            type="button"
+            className={buttonReset}
+            onClick={() => {
+              setSortType(SortType.NONE);
+              setIsReversed(false);
+            }}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {reorderedGoods.map((good, index) => (
+            <li key={index} data-cy="Good">
```

Comment:
Again, the key is being used as an index 😢. I believe I mentioned this in a yours other task. Please use unique values instead. It's not a problem in this case, though.

Here is article why key is important https://www.epicreact.dev/why-react-needs-a-key-prop